bokeh/embed.py

```diff
@@ -131,10 +132,37 @@ def components(models, wrap_script=True, wrap_plot_info=True):
             values.append(models[k])
         models = values
 
-    # 2) Do our rendering
-
-    with _ModelInDocument(models):
-        (docs_json, render_items) = _standalone_docs_json_and_render_items(models)
+    # 2) Append models to one document. Either pre-existing or new.
+    doc = None
+    for model in models:
+        if isinstance(model, Document):
+            doc = model
+            break
+        elif isinstance(model, Model):
+            if model.document is not None:
+                doc = model.document
+                break
+    if doc is None:
+        # oh well - just make up a doc
+        doc = Document()
```

Comment:
I think it would be good to assert that if any of the models have a doc, then they all do, and also that they are all the same. This could look something like:
```
existing_docs = set(m.document for m in models if m.document is not None)
if len(existing_docs) == 0:
    doc = Document()
else if len(existing_doc) == 1:
    doc = existing_dos.pop()
else:
    # conflicting/multiple docs, raise an error
```

Regarding the `if isinstance(model, Document)` part... is that something that is supported currently? I am not sure what it would mean to pass e.g. two documents and three models in... I don't think we can do anything sensible with that? In which case it might make sense to either not allow documents to be passed in, or to only allow *one* to be passed in, e.g.:
```
if any(isinstance(model, Document) for model in models) and len(models) > 1:
    # raise an error
```
But if we allow this we might also want to come up with a better parameter name than `models` since it seems to not be accurate.